Include/Rover/LinearRegressionModel.hpp

```diff
@@ -9,14 +9,20 @@ namespace Rover {
   //! Models a trial using linear regression.
   /*!
     \tparam T The Trial to model.
+    \tparam C The arithmetic type used for calculations.
+    \detail Every argument and the result of the Trial must be convertible to C
+    \detail C must be convertible to Trial's result
   */
-  template<typename T>
+  template<typename T, typename C = double>
```

Comment:
On Intel, 64-bit floats are most commonly used for BLAS as the performance overhead is quite low (less than x2) but produce much more stable LU decomposition.